rpc/lib/client/http_client.go

```diff
@@ -25,7 +25,10 @@ type HTTPClient interface {
 }
 
 // TODO: Deprecate support for IP:PORT or /path/to/socket
-func makeHTTPDialer(remoteAddr string) (string, func(string, string) (net.Conn, error)) {
+func makeHTTPDialer(remoteAddr string) (string, string, func(string, string) (net.Conn, error)) {
```

Comment:
Seems like the simplest solution, to add the http/https protocol here.
This is not used anywhere else than the below points, right? (Being private doesn't seem a bad change).